hybridse/src/udf/default_udf_library.cc

```diff
@@ -310,6 +310,47 @@ struct DistinctCountDef {
     };
 };
 
+template <typename T>
+struct MedianDef {
+    using ArgT = typename DataTypeTrait<T>::CCallArgType;
+    using VectorT = std::vector<T>;
+
+    void operator()(UdafRegistryHelper& helper) {
+        std::string suffix = ".opaque_vector_" + DataTypeTrait<T>::to_string();
+        helper.templates<Nullable<double>, Opaque<VectorT>, Nullable<T>>()
+            .init("median_init" + suffix, MedianDef::Init)
+            .update("median_update" + suffix, MedianDef::Update)
+            .output("meadin_output" + suffix, reinterpret_cast<void*>(MedianDef::Output), true);
+    }
+
+    static void Init(VectorT* addr) { new (addr) VectorT(); }
+
+    static VectorT* Update(VectorT* vec, T value, bool is_null) {
+        if (!is_null) {
+            vec->push_back(value);
+        }
+        return vec;
+    }
+
+    static void Output(VectorT* vec, double* ret, bool* is_null) {
+        if (vec->empty()) {
+            *is_null = true;
+        } else {
+            *is_null = false;
+
+            std::sort(vec->begin(), vec->end());
```

Comment:
using two heap structure in update help avoid sort

Reply:
> using two heap structure in update help avoid sort

Wow, good idea.